### PR TITLE
Fix `pex3 lock export`: re-use `--lock` resolver.

### LINKED ISSUE #1645: Clarify `pex3 lock export` command.
Both #1404 and #1405 are difficult problems to solve in order to have a `pex3 lock export` command that works without caveats (i.e.: that exports a universal lock as a universal requirements.txt). The `pex3 lock export` command should be better documented and possibly modified to explicitly support only exporting to one (or maybe more) explicit local interpreters or foreign platforms.

### LINKED ISSUE #1826: `pex3 lock export` does't seem to respect the `platform` flag.
context: 
The [docker](https://github.com/docker/docker-py) python package has a [windows only dependency on pywin32](https://github.com/docker/docker-py/blob/a48a5a9647761406d66e8271f19fab7fa0c5f582/requirements.txt#L14).

repo steps:
```shell
pex3 lock create --style universal --resolver-version pip-2020-resolver docker -o lock.json
pex3 lock export --platform current lock.json
```

output:
```
certifi==2022.6.15 \
    --hash=sha256:fe86415d55e84719d75f8b69414f6438ac3547d2078ab91b67e779ef69378412  \
    --hash=sha256:84c85a9078b11105f04f3036a9482ae10e4621616db313fe045dd24743a0820d
charset-normalizer==2.1 \
    --hash=sha256:5189b6f22b01957427f35b6a08d9a0bc45b46d3788ef5a92e978433c7a35f8a5  \
    --hash=sha256:575e708016ff3a5e3681541cb9d79312c416835686d054a23accb873b254f413
docker==5.0.3 \
    --hash=sha256:7a79bb439e3df59d0a72621775d600bc8bc8b422d285824cb37103eab91d1ce0  \
    --hash=sha256:d916a26b62970e7c2f554110ed6af04c7ccff8e9f81ad17d0d40c75637e227fb
idna==3.3 \
    --hash=sha256:84d9dd047ffa80596e0f246e2eab0b391788b0503584e8945f2368256d2735ff  \
    --hash=sha256:9d643ff0a55b762d5cdb124b8eaa99c66322e2157b69160bc32796e824360e6d
pywin32==227 \
    --hash=sha256:f27cec5e7f588c3d1051651830ecc00294f90728d19c3bf6916e6dba93ea357c  \
    --hash=sha256:47a3c7551376a865dd8d095a98deba954a98f326c6fe3c72d8726ca6e6b15507  \
    --hash=sha256:a929a4af626e530383a579431b70e512e736e9588106715215bf685a3ea508d4  \
    --hash=sha256:371fcc39416d736401f0274dd64c2302728c9e034808e37381b5e1b22be4a6b0  \
    --hash=sha256:c054c52ba46e7eb6b7d7dfae4dbd987a1bb48ee86debe3f245a2884ece46e295  \
    --hash=sha256:7f18199fbf29ca99dff10e1f09451582ae9e372a892ff03a28528a24d55875bc  \
    --hash=sha256:4cdad3e84191194ea6d0dd1b1b9bdda574ff563177d2adf2b4efec2a244fa116  \
    --hash=sha256:31f88a89139cb2adc40f8f0e65ee56a8c585f629974f9e07622ba80199057511  \
    --hash=sha256:7c1ae32c489dc012930787f06244426f8356e129184a02c25aef163917ce158e  \
    --hash=sha256:f4c5be1a293bae0076d93c88f37ee8da68136744588bc5e2be2f299a34ceb7aa  \
    --hash=sha256:300a2db938e98c3e7e2093e4491439e62287d0d493fe07cce110db070b54c0be  \
    --hash=sha256:9b31e009564fb95db160f154e2aa195ed66bcc4c058ed72850d047141b36f3a2
requests==2.28.1 \
    --hash=sha256:8fefa2a1a1365bf5520aac41836fbee479da67864514bdb821f31ce07ce65349  \
    --hash=sha256:7c5599b102feddaa661c826c56ab4fee28bfd17f5abca1ebbe3e7f19d7c97983
urllib3==1.26.9 \
    --hash=sha256:44ece4d53fb1706f667c9bd1c648f5469a2ec925fcf3a776667042d645472c14  \
    --hash=sha256:aabaf16477806a5e1dd19aa41f8c2b7950dd3c746362d7e3223dbe6de6ac448e
websocket-client==1.3.3 \
    --hash=sha256:5d55652dc1d0b3c734f044337d929aaf83f4f9138816ec680c1aefefb4dc4877  \
    --hash=sha256:d58c5f284d6a9bf8379dab423259fe8f85b70d5fa5d2916d5791a84594b122b1
```

I expect the output to not include `pywin32`

### PR DESCRIPTION
**Pull request by jsirois**
Previously the lock export utility was using out-dated logic from the
early days of the lock project. Extract a `subset` operation from the
lock resolver and have it and lock export share that code.

Also properly plumb lock export target selection and validation of
exactly one target with updated help and error messages to make it clear
that locks can only be exported for a single target.

Fixes #1826
Closes #1645

### CODE REVIEW

File: pex/resolve/lockfile/subset.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    network_configuration=None,  # type: Optional[NetworkConfiguration]
+    build=True,  # type: bool
+    use_wheel=True,  # type: bool
+    prefer_older_binary=False,  # type: bool
+    transitive=True,  # type: bool
+    include_all_matches=False,  # type: bool
+):
+    # type: (...) -> Union[SubsetResult, Error]
+
+    with TRACER.timed("Parsing requirements"):
```
Comment (jsirois) [inline pex/resolve/lockfile/subset.py:56]: This was extracted from the lock_resolver here: https://github.com/pantsbuild/pex/pull/1831/files#diff-aad7222a03ff8182eb9c406092214ea24c7811b23dbdb9f47bfedc85aab327d0L240

File: pex/resolve/lockfile/model.py [hunk 2]
```diff
@@ -123,39 +115,3 @@ def extract_requirement(req):
     locked_resolves = attr.ib()  # type: SortedTuple[LockedResolve]
     local_project_requirement_mapping = attr.ib(eq=False)  # type: Mapping[str, Requirement]
     source = attr.ib(default=None, eq=False)  # type: Optional[str]
-
-    def select(self, targets):
```
Comment (jsirois) [inline pex/resolve/lockfile/model.py:127]: This was the entry-point to the out-dated lock selection logic export (and update) were using. Now all removed.

File: pex/cli/commands/lock.py [hunk 3]
```diff
@@ -475,17 +506,34 @@ def _update(self):
...
+                    use_wheel=lock_file.allow_wheels,
+                    prefer_older_binary=lock_file.prefer_older_binary,
+                    transitive=lock_file.transitive,
+                )
+            )
+
         update_requests = [
-            ResolveUpdateRequest(target=target, locked_resolve=locked_resolve)
-            for target, locked_resolve in lock_file.select(targets.unique_targets())
+            ResolveUpdateRequest(
```
Comment (jsirois) [inline pex/cli/commands/lock.py:533]: And this was the flip-over from the out-dated lock selection logic for `lock update` which came along for the ride so I could nuke the outdated logic wholesale. The existing update test still pass, vetting this.

**Review by asherf — APPROVED** [review]

**Comment by jsirois** [comment]
Heads up that I'll need 1 more review. Asher took a look, but I'll need maintainer sign off too.

**Review by benjyw — APPROVED** [review]

**Pull request merged**